File: backend/app/models/ticket.py
# SQLAlchemy ORM model: tickets table
# Columns (from CSV + NLP enrichment + assignment):
#   id (UUID PK), session_id (str, indexed), customer_guid, gender, date_of_birth,
#   segment (Mass/VIP/Priority), description, attachments,
#   country, region, city, street, building_number,
#   latitude (Float nullable), longitude (Float nullable),
#   request_type, sentiment, priority_score (Int 1-10), language (KZ/ENG/RU), summary,
#   assigned_manager_id (FK → managers.id nullable),
#   assigned_office_id (FK → business_units.id nullable),
#   created_at (DateTime default now)

File: frontend/src/types/ticket.ts
export type RequestType =
  | 'Complaint'
  | 'DataChange'
  | 'Consultation'
  | 'Claim'
  | 'AppMalfunction'
  | 'FraudulentActivity'
  | 'Spam'

export type Sentiment = 'Positive' | 'Neutral' | 'Negative'

export type Language = 'KZ' | 'ENG' | 'RU'

export type Segment = 'Mass' | 'VIP' | 'Priority'

export interface Ticket {
  id: string
  customerGuid: string
  gender: string
  dateOfBirth: string
  segment: Segment
  description: string
  attachments: string
  country: string
  region: string
  city: string
  street: string
  buildingNumber: string
  latitude: number | null
  longitude: number | null
  // NLP enrichment
  requestType: RequestType
  sentiment: Sentiment
  priorityScore: number
  language: Language
  summary: string
  // Assignment
  assignedManagerId: string | null
  assignedManagerName: string | null
  assignedOfficeId: string | null
  assignedOfficeName: string | null
  sessionId: string
  createdAt: string
}
